src/services/array-functions.js

```diff
@@ -1,35 +1,56 @@
 
 //in the function map, create a new array and store in a variable
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop into the call to fnc
 //add the returned value from fnc to the new array
 //return the new array
 export function map(theArray, fnc){
-
+  let newArray = [];
+    for (let i = 0 ; i < theArray.length ; i++){
+  let map1 = theArray[i]
+  let map2 = fnc(map1);
+  newArray[i] = (map2);
+
+  }
+   return newArray;
 }
 
+
 //create a new array
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
 //fnc will return true or false, if true add the item to the new array else do not
 //return the new array
 export function filter(theArray, fnc){
-
+  let filtterArray = [];
+  for (var i = 0; i < theArray.length; i++ ) {
+    if (fnc(theArray[i])) {
+       filtterArray.push(theArray[i]);
+    }
+  }
+  return filtterArray;
 }
 
 
-//loop theArray and call the fnc for each thing in the array, 
+//loop theArray and call the fnc for each thing in the array,
 //     passing in the item from the current loop
-//fnc will return true or false, if true return the item 
+//fnc will return true or false, if true return the item
 //return null
 export function find(theArray, fnc){
+  let findArray = [];
```

Comment:
Look over this again. The instructions don't tell you to make an array or push to an array